src/log-entry-dashboard/log-entry-form/log-entry-form.component.tsx

```diff
@@ -20,124 +20,305 @@ import {
 import { withAuthenticationContext } from '../../_shared/contexts/authentication/with-authentication-context.component';
 import { LogEntry } from '../../_shared/interfaces/log-entry.interface';
 import { createLogEntry } from '../../_shared/services/log-entry.service';
-import { inputClass } from './log-entry-form.styles';
-import { panelClass } from '../log-entry-dashboard.styles';
+import { timeForm, buttonContainer } from './log-entry-form.styles';
+import { RadioChangeEvent } from 'antd/lib/radio';
+import { any } from 'prop-types';
 
 const FormItem = Form.Item;
+const RadioButton = Radio.Button;
+const RadioGroup = Radio.Group;
+const { TextArea } = Input;
+const SelectOption = Select.Option;
 
 interface LogEntryFormProps extends FormComponentProps {
   authContext: IAuthenticationContext;
 }
 
 class LogEntryFormComponent extends React.Component<LogEntryFormProps> {
+  state = {
+    categoryId: 0,
+    pointsCompeleted: 'notCompleted',
+  };
+
+  handleHourChange = (e?: number) => {
+    const {
+      form: { setFieldsValue },
+    } = this.props;
+
+    setFieldsValue({ logValue: 0 });
+  };
+
+  handleCategoryChange = (e: string) => {
+    this.setState({ categoryId: e });
+  };
+
+  handleRaidoChange = (e: RadioChangeEvent) => {
+    if (e.target.value === 'completed') {
+      this.setState({ pointsCompeleted: 'completed' });
+    } else if (e.target.value === 'notCompleted') {
+      this.setState({ pointsCompeleted: 'notCompleted' });
+    }
+  };
+
+  hasErrors = (fieldsError: any) => {
```

Comment:
is this necessary?  doesn't `getFieldErrors` return null or something if there are no errors?  I don't recall and haven't looked at the docs so just asking.

Reply:
This function is deactivating the submit button when errors are present. I am not sure if there is a more elegant way to do it.